### ISSUE TITLE
Inconsistent TypeScript definitions in the module @keycloak/keycloak-admin-client while compiling

### ISSUE THREAD

**Issue by igorlino** [keycloak/keycloak#25085]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/client/node

### Describe the bug

After updating to @keycloak/keycloak-admin-client/23.0.0
i get compilation errors due to inconsistent type definitions in the admin client module.

### Version

23.0.0

### Expected behavior

compilation has no TypeScript errors, TS definitions are correct.

### Actual behavior

```
❯ npm run build

> test-service@0.1.0 build
> tsc

src/test/node_modules/@keycloak/keycloak-admin-client/lib/resources/agent.d.ts:18:15 - error TS2304: Cannot find name 'HeadersInit'.

18     headers?: HeadersInit;
                 ~~~~~~~~~~~

src/test/node_modules/@keycloak/keycloak-admin-client/lib/utils/fetchWithError.d.ts:10:47 - error TS2552: Cannot find name 'RequestInfo'. Did you mean 'RequestInit'?

10 export declare function fetchWithError(input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
                                                 ~~~~~~~~~~~
``` 

### How to Reproduce?

create any empty project and include @keycloak/keycloak-admin-client as dependency.
npm install

Open in Visual Studio Code the project, and open both files
node_modules/@keycloak/keycloak-admin-client/lib/resources/agent.d.ts:18:15 
node_modules/@keycloak/keycloak-admin-client/lib/utils/fetchWithError.d.ts:10:47

See errors in the editor

### Anything else?

This became visible after I migrated from the previous deprecated package "keycloak-admin": "^1.14.18".

**Comment by jonkoops** [comment]
You'll have to set your [`lib`](https://www.typescriptlang.org/tsconfig#lib) option to a more modern version.

**Comment by jonkoops** [comment]
Ah yes, that makes sense. I forgot those types were contained in the `DOM` library. Good to hear you managed to solve the problem.

**Comment by adelisle** [comment]
This bug isn't fixed, requiring DOM is not the solution.

**Comment by jonkoops** [comment]
I am sure the Node.js types will have this at this point as well, as the Fetch API is supported there. Make sure your `@types/node` dependency is up to date and properly loaded.

**Comment by Yogu** [comment]
> I am sure the Node.js types will have this at this point as well, as the Fetch API is supported there

The problem is not that Node.js did not include the Fetch API. The problem is that keycloak is relying on an exact detail of how the Fetch API is typed in the `dom` lib of typescript; precisely, that it includes this type alias:

```
type RequestInfo = Request | string;
```

`@types/node` simply does not declare this type but instead inlines `Request | string`. That's what keycloak can do as well: Use the type `Request | string` instead of `RequestInfo`.

The type alias `RequestInfo` is based on a [typedef of the same name in the spec](https://fetch.spec.whatwg.org/#requestinfo), so maybe it would also be ok to add it to the node types. I created a PR for that: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71424

**Comment by Yogu** [comment]
I failed with the PR, at least unless someone brings in a new idea. I think the reason `@types/node` does not have any type aliases like `RequestInfo` is that it needs to work both with and without the `dom` lib, and if it redeclares the type alias, it cannot work with the lib.

-> The only way to fix this is probably in keycloak. Do you accept a PR where I remove the `dom` lib and inline the type alias?

**Comment by jonkoops** [comment]
Yeah, we would totally accept a PR to fix that if you're interested in landing that. Let me re-open this issue as I agree this should work without having to rely on the `dom` types in a Node.js context.

**Comment by Yogu** [comment]
submitted #35968 with a fix
